src/pages/PostVideo/PostVideo.jsx

```diff
@@ -0,0 +1,287 @@
+import React, { useReducer, useState } from "react";
+import bannerStyles from "../Home/Home.module.css";
+import loginStyles from "../Login/Login.module.css";
+import spinner from "../../assets/spinner.gif";
+import { useNavigate, useLocation } from "react-router-dom";
+import styles from "./PostVideo.module.css";
+import { BsPlus, BsCloudUpload } from "react-icons/bs";
+import { useData } from "../../context";
+import { initialVideoCreds, reducer } from "./reducer";
+import { getImageLink, postVideo } from "../../services";
+export const PostVideo = () => {
+  const { categories, languages, dispatch: dataDispatcher } = useData();
+  const [state, dispatch] = useReducer(reducer, initialVideoCreds);
+  const [actorInput, setActorInput] = useState("");
+  const [uploading, setUploading] = useState(false);
+
+  const navigator = useNavigate();
+
+  const submitHandler = async (e) => {
+    e.preventDefault();
+    const youtubeId = state._id
+      .replace("https://www.youtube.com/watch?v=", "")
+      .replace("https://youtu.be/", "");
+
+    const resp = await postVideo({ ...state, _id: youtubeId });
+    if (resp.status === "success") {
+      dataDispatcher({
+        type: "ADD_VIDEO",
+        payload: resp.video,
+      });
+      dispatch({ type: "RESET" });
+      navigator("/explore");
+    }
+  };
+
+  const imageHandler = async (e) => {
+    setUploading(true);
+    const imageLink = await getImageLink(e);
+    console.log(imageLink);
+    dispatch({ type: "SET_THUMBNAIL", payload: imageLink });
+    setUploading(false);
+  };
```

Comment:
You can do this when you click on upload video, in onchange update the value in state.
In this way you will not upload unwanted images

Reply:
Thanks for this, its a great suggestion.